Fix #5149: Items disappearing from shulkers

Xicx99 opened the pull request:
The item wand of symmetry was not handling the NBT information from the items. 
The fix consisted on making sure to also take into account this information when looking for the item in the inventory, when placing the new item and when removing the item.
Created a simple test to confirm the fix.

Xicx99 commented:
Hello,
Since Mojang changed the way NBT tags work from version 1.20.1 to 1.21.1, I can't simply change the target branch because the code wouldn't work.
Should I update the code to the right version and open a new PR?

VoidLeech commented:
Opening a new PR is fine. Alternatively you could make changes from 1.20.1 and force-push that to this branch.

RaymondBlaze commented on src/main/java/com/simibubi/create/foundation/utility/BlockHelper.java:
```diff
@@ -122,7 +123,7 @@ public static BlockState setZeroAge(BlockState blockState) {
 	}
 
-	public static int findAndRemoveInInventory(BlockState block, Player player, int amount) {
+	public static int findAndRemoveInInventory(BlockState block, String nbtString, Player player, int amount) {
```
Is there any specific reason for using `String` here instead of just pass in the `ItemStack` and use  `ItemStack#isSameItemSameComponents`?

Also please don't remove the original method, leave it as an overload so no breaking changes are introduced.

Xicx99 replied:
I was simply not aware that this method existed...
Now developing for 1.20.1 I will change that and use the equivalent method for it `ItemStack#isSameItemSameTags`.

Thank you for pointing these out.

Xicx99 committed:
Fix #5149: Items disappearing from shulkers
The item wand of symmetry was not handling
the NBT information from the items.
The fix consisted on making sure to also take into account
this information when placing and removing the items.

Xicx99 commented:
@VoidLeech 
Hello, I've force-pushed the code for 1.20.1 to here and changed the target branch.
Hopefully everything is good now.
Thank you.

VoidLeech commented on src/main/java/com/simibubi/create/foundation/utility/BlockHelper.java:
```diff
@@ -183,6 +183,78 @@ public static int findAndRemoveInInventory(BlockState block, Player player, int
+				.getItem(preferredSlot);
+			int count = itemstack.getCount();
+			if (itemstack.getItem() == required && count > 0) {
+				if (wantedStack == null || ItemStack.isSameItemSameTags(wantedStack, itemstack)) {
```
Since this new method (is the old one still used anywhere) seems to copy most of the original, and does handle a null value for the new parameter, you should have the original call this method instead.

Also, does `ItemStack.isSameItemSameTags` cover item capabilities? For example, with cold sweat, you can add insulation to armor (like the backtank, which you can place), which gets saved via capabilities instead of direct nbt on the stack.

Xicx99 replied:
Firstly, just to make sure I got it right, I should have the original method call the new one with the parameter `ItemStack wantedStack` as null?

From my understanding `ItemStack.isSameItemSameTags`  does not cover capabilities, it just compares the NBT tags. It might require additional checks.
